.github/workflows/comment-on-pr-with-build-url.yml

```diff
@@ -0,0 +1,29 @@
+name: Comment On PR With Build URL
+on:
+  workflow_run:
+    workflows: ['Slash Command Build']
+    types: [completed]
+jobs:
+  commentOnPrWithBuildUrl:
+    if: github.event.workflow_run.event == 'repository_dispatch' && github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Fetch artifact
+        id: fetch_artifact
+        run: |
+          artifact=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${{github.repository}}/actions/runs/${{github.event.workflow_run.id}}/artifacts | jq .artifacts[0])
+          artifact_id=$(jq -r '.id' <<< "${artifact}")
+          artifact_name=$(jq -r '.name' <<< "${artifact}")
+          issue_id=${artifact_name##*-}
```

Comment:
⚠️ This is a bit of a hack. I am extracting the pull request ID from the artifact name. This is necessary because there is seemingly no way of passing this information from the previous workflow (`slash-command-build.yml`).

i.e. the name of the artifact is something like `woocommerce-payments-pr-100` with `100` being the issue/PR ID.

Reply:
Hopefully we can find someone with GitHub actions experience to give us some other options here – sounds a little fragile.

Maybe take a step back and look at other ways to make an artifact available. There might be other approaches (as opposed to commenting on the PR) that work well out of the box?

Another question - is this use-case pushing the limits of GitHub actions, and we should really be using a "real" CI system?

Reply:
So you're pulling the artifact id out of the github api (based on the issue/pr_id) and then passing that artifact id to a 3rd party service which redirects (?) to the artifact direct download link?

Reply:
> Hopefully we can find someone with GitHub actions experience to give us some other options here – sounds a little fragile.

It's not ideal but it's not overly fragile, we're fully in control of the artifact name. I can't think of a way that this would break other than the API changing.

> Maybe take a step back and look at other ways to make an artifact available. There might be other approaches (as opposed to commenting on the PR) that work well out of the box?

The artifact is available from [actions page](https://github.com/Automattic/woocommerce-payments/actions) but you'd need to navigate to the specific workflow run to find it. A PR comment seems like the best UX as it keeps everything inline in the PR itself but I'm definitely open to other ideas.

> Another question - is this use-case pushing the limits of GitHub actions

It really seems like it. There are a number of hacks in this implementation that I'm not super happy with. It seems like the GitHub artifact API was not really intended for this use case.

> and we should really be using a "real" CI system

Probably yes, but I wanted to keep the implementation lean and see if was possible to achieve the desired outcome using GitHub workflows.

> So you're pulling the artifact id out of the github api (based on the issue/pr_id) and then passing that artifact id to a 3rd party service which redirects (?) to the artifact direct download link?

Yep, that's exactly right.